homeassistant/components/smartthings/fan.py

```diff
@@ -41,19 +41,50 @@ async def async_setup_entry(
 
 def get_capabilities(capabilities: Sequence[str]) -> Sequence[str] | None:
     """Return all capabilities supported if minimum required are present."""
-    supported = [Capability.switch, Capability.fan_speed]
-    # Must have switch and fan_speed
-    if all(capability in capabilities for capability in supported):
-        return supported
-    return None
+
+    # MUST support switch as we need a way to turn it on and off
+    if Capability.switch not in capabilities:
+        return None
+
+    # These are all optional but at least one must be supported
+    optional = [
+        Capability.air_conditioner_fan_mode,
+        Capability.fan_speed,
+    ]
+
+    # If none of the optional capabilities are supported then error
```

Comment:
What's the error?

Reply:
Returns null to indicate that the fan platform cannot be supported. The higher level then moves on to the next platform.

Reply:
Ok, but that doesn't sound like an error. Please clarify the comment with your first sentence in your reply here instead.